src/new-client/src/App/Trades/TradesState/filterState.ts

```diff
@@ -21,35 +22,75 @@ export const fieldValuesContainer = Object.freeze(
 const ClonedFieldValuesContainer = () =>
   mapObject(fieldValuesContainer, () => new Set()) as DistinctValues
 
-export const [useDistinctFieldValues, distinctFieldValues$] = bind(
-  trades$.pipe(
-    map((trades) =>
-      trades.reduce((distinctValues, trade) => {
-        for (const field in trade) {
-          ;(distinctValues[field as keyof Trade] as Set<unknown>).add(
-            trade[field as keyof Trade],
-          )
-        }
-        return distinctValues
-      }, ClonedFieldValuesContainer()),
-    ),
+const distinctValues$ = trades$.pipe(
+  map((trades) =>
+    trades.reduce((distinctValues, trade) => {
+      for (const field in trade) {
+        ;(distinctValues[field as keyof Trade] as Set<unknown>).add(
+          trade[field as keyof Trade],
+        )
+      }
+      return distinctValues
+    }, ClonedFieldValuesContainer()),
   ),
 )
 
-export const [colFilterSelections$, onColFilterSelect] = createListener<
-  [ColField, Set<unknown>]
->()
+export const [useDistinctFieldValues, distinctFieldValues$] = bind(
+  (field: ColField) =>
+    distinctValues$.pipe(map((distinctValues) => distinctValues[field])),
+  new Set(),
+)
 
-export const [useAppliedFilters, appliedFilters$] = bind(
-  colFilterSelections$.pipe(
-    scan((appliedFilters, [field, filterSet]) => {
-      return {
-        ...appliedFilters,
-        [field]: filterSet,
-      }
-    }, ClonedFieldValuesContainer()),
-    startWith(ClonedFieldValuesContainer()),
-  ),
+abstract class FilterEvent {
+  constructor(public colField: ColField) {}
+}
```

Comment:
Why we need to create abstract class here? To accept parameters?

Reply:
We don't need to, but I think it helps define the interface.  Every column filter event is going to need a column field and something else (in the case of the reset event, you don't need another piece of data, but you need to know you are resetting).  So this lets us organize all the filter events as extending this class (and forex define a function that takes this type as input), but the class itself will never be directly instantiated.

In terms of why use classes at all... I realize we don't do much OOP in this project.  But, with TypeScript IME, it makes it easier to define a big switch working with _instanceof_ .